uint8 index
int32 bias
---